devices/IWearRemapper/src/IWearRemapper.cpp

```diff
@@ -946,7 +989,8 @@ IWearRemapper::impl::getSensor(const sensor::SensorName name,
                                const sensor::SensorType type,
                                std::map<std::string, SensorPtr<SensorImpl>>& storage)
 {
-    if (!validSensorName(name, type)) {
+    // TODO check if this check on the sensor name is required, and how it is done in case rpc is not available
+    if (useRPC && !validSensorName(name, type)) {
```

Comment:
Let's try to recap the situation about the wearable status. The remapper can attach to many wearable sources, each of them providing a set of sensors. The remapper is ready only when all it has received at least one data sample from all the sources.

The callback is executed on all received wearable data messages. This means that if you have 2 sources, but only one is streaming, the overall wearable status of the remapper should still be `WaitingForFirstData`.

All the devices attached to the remapper during their attach step wait that the `IWear` interface exposed by the remapper is `Ok`, e.g.:

https://github.com/robotology/human-dynamics-estimation/blob/ef728292b4fd7b4ba62a50fafd7014c92853ebed/devices/HumanStateProvider/HumanStateProvider.cpp#L624-L633

This means that they wait that at least one data sample is received, and consequently all the sensors are already known. This makes me think that the RPC logic is not anymore useful. It would be useful only if the attached devices would ask the sensors _before_ the remapper is ready.

@lrapetti @Yeshasvitvs does this make sense? Am I missing something?

Reply:
> This means that they wait that at least one data sample is received, and consequently all the sensors are already known. This makes me think that the RPC logic is not anymore useful. It would be useful only if the attached devices would ask the sensors _before_ the remapper is ready.

I confirm this is the case. For example when using the `humanStateProvider`, it remains in the `WaitingForFirstData` state unitil the first data is streamed, both in the case of using `rpc` and without the `rpc`.

This makes the `rpc` port unnecessary for this type of usage, which I would say is possibily the one we will be most likely using. So, if we can define this as being the ''standard'' usage, then we can choose to have `useRPC` as optional parameter in case one explicitly want to use the `rpc` port (as stated by @diegoferigo here: https://github.com/robotology-playground/wearables/pull/24#discussion_r255095740).

Reply:
Great! So it is not strictly necessary. At this point, since this PR has other improvements and I don't want to block it, can you please open a new issue for discussing if we want to maintain this RPC support (that complicates things) or we just have to document this behavior stating how to handle consumer devices as it currently happens in our own ones?

Reply:
The issue to discuss if maintain `RPC` support has been opened at https://github.com/robotology-playground/wearables/issues/28.

Reply:
Great, thanks!